src/services/completions.ts

```diff
@@ -826,10 +832,20 @@ namespace ts.Completions {
         log("getCompletionData: Is inside comment: " + (timestamp() - start));
 
         if (insideComment) {
-            // The current position is next to the '@' sign, when no tag name being provided yet.
-            // Provide a full list of tag names
-            if (hasDocComment(sourceFile, position) && sourceFile.text.charCodeAt(position - 1) === CharacterCodes.at) {
-                isJsDocTagName = true;
+            if (hasDocComment(sourceFile, position)) {
+                // The current position is next to the '@' sign, when no tag name being provided yet.
+                // Provide a full list of tag names
+                if (sourceFile.text.charCodeAt(position - 1) === CharacterCodes.at) {
+                    isJsDocTagName = true;
+                }
+                else {
+                    const lineStart = getLineStartPositionForPosition(position, sourceFile);
+                    shouldAppendAtSignBeforeJsDocTagName = sourceFile.text.substr(lineStart, position).indexOf("@") === -1;

```

Comment:
@mhegazy Good point.  I agree above check will allow tagname to be in the middle of the comment...though check only for "*" and whitespace may not be enough. Consider following example:
```js
/**
 |completion
*/

/**
          |completion|
 */

/**
  *       |completion|
  */
```